AssertionError: found

With Cthulhu v2.8.2 and Julia 1.9.0-rc1:

```julia
julia> using Cthulhu

julia> struct Foo
           a
           b
       end

julia> function bar(foo)
           a, (b1, b2) = foo.a, foo.b
           return b1
       end
bar (generic function with 1 method)

julia> bar(Foo(2, (3, 4)))
3

julia> @descend bar(Foo(2, (3, 4)))
ERROR: AssertionError: found
Stacktrace:
  [1] map_ssas_to_source(src::Core.CodeInfo, rootnode::JuliaSyntax.TreeNode{JuliaSyntax.SyntaxData}, Δline::Int64)
    @ TypedSyntax ~/.julia/packages/TypedSyntax/xKbe0/src/node.jl:510
  [2] map_ssas_to_source
    @ ~/.julia/packages/TypedSyntax/xKbe0/src/node.jl:598 [inlined]
  [3] tsn_and_mappings(m::Method, src::Core.CodeInfo, rt::Any; warn::Bool, strip_macros::Bool, kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ TypedSyntax ~/.julia/packages/TypedSyntax/xKbe0/src/node.jl:44
  [4] tsn_and_mappings
    @ ~/.julia/packages/TypedSyntax/xKbe0/src/node.jl:28 [inlined]
  [5] #get_typed_sourcetext#32
    @ ~/.julia/packages/Cthulhu/e6txW/src/reflection.jl:350 [inlined]
  [6] get_typed_sourcetext
    @ ~/.julia/packages/Cthulhu/e6txW/src/reflection.jl:348 [inlined]
  [7] find_callsites(interp::Cthulhu.CthulhuInterpreter, CI::Core.CodeInfo, stmt_infos::Vector{Core.Compiler.CallInfo}, mi::Core.MethodInstance, slottypes::Vector{Any}, optimize::Bool, annotate_source::Bool)
    @ Cthulhu ~/.julia/packages/Cthulhu/e6txW/src/reflection.jl:28
  [8] _descend(term::REPL.Terminals.TTYTerminal, interp::Cthulhu.CthulhuInterpreter, curs::Cthulhu.CthulhuCursor; override::Nothing, debuginfo::Symbol, optimize::Bool, interruptexc::Bool, iswarn::Bool, hide_type_stable::Bool, verbose::Nothing, remarks::Bool, with_effects::Bool, inline_cost::Bool, type_annotations::Bool, annotate_source::Bool)
    @ Cthulhu ~/.julia/packages/Cthulhu/e6txW/src/Cthulhu.jl:496
  [9] _descend
    @ ~/.julia/packages/Cthulhu/e6txW/src/Cthulhu.jl:413 [inlined]
 [10] #_descend#107
    @ ~/.julia/packages/Cthulhu/e6txW/src/Cthulhu.jl:765 [inlined]
 [11] _descend
    @ ~/.julia/packages/Cthulhu/e6txW/src/Cthulhu.jl:765 [inlined]
 [12] #_descend#110
    @ ~/.julia/packages/Cthulhu/e6txW/src/Cthulhu.jl:781 [inlined]
 [13] __descend_with_error_handling(args::Any; terminal::Any, kwargs::Base.Pairs{Symbol, V, Tuple{Vararg{Symbol, N}}, NamedTuple{names, T}} where {V, N, names, T<:Tuple{Vararg{Any, N}}})
    @ Cthulhu ~/.julia/packages/Cthulhu/e6txW/src/Cthulhu.jl:239
 [14] _descend_with_error_handling(f::Any, argtypes::Any; kwargs::Base.Pairs{Symbol, Bool, Tuple{Symbol}, NamedTuple{(:iswarn,), Tuple{Bool}}})
    @ Cthulhu ~/.julia/packages/Cthulhu/e6txW/src/Cthulhu.jl:228
 [15] descend_code_typed(::Any, ::Vararg{Any}; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Cthulhu ~/.julia/packages/Cthulhu/e6txW/src/Cthulhu.jl:173
 [16] descend_code_typed(::Any, ::Any)
    @ Cthulhu ~/.julia/packages/Cthulhu/e6txW/src/Cthulhu.jl:173
 [17] top-level scope
    @ REPL[5]:1
```

--- a/TypedSyntax/Project.toml
+++ b/TypedSyntax/Project.toml
@@ -1,7 +1,7 @@
 name = "TypedSyntax"
 uuid = "d265eb64-f81a-44ad-a842-4247ee1503de"
 authors = ["Tim Holy <tim.holy@gmail.com> and contributors"]
-version = "1.0.9"
+version = "1.0.10"
 
 [deps]
 CodeTracking = "da1fd8a2-8d9e-5ec2-8556-3022fb5608a2"

--- a/TypedSyntax/src/node.jl
+++ b/TypedSyntax/src/node.jl
@@ -495,21 +495,10 @@ function map_ssas_to_source(src::CodeInfo, rootnode::SyntaxNode, Δline::Int)
                         while !is_prec_assignment(lhsnode) && lhsnode.parent !== nothing
                             lhsnode = lhsnode.parent
                         end
-                        lhsnode = child(lhsnode, 1)
-                        if kind(lhsnode) == K"tuple"   # tuple destructuring
-                            found = false
-                            for child in children(lhsnode)
-                                if kind(child) == K"Identifier"
-                                    if child.val == sym
-                                        lhsnode = child
-                                        found = true
-                                        break
-                                    end
-                                end
-                            end
-                            @assert found
+                        lhsnode, found = find_identifier_or_tuplechild(child(lhsnode, 1), sym)
+                        if found
+                            symtyps[lhsnode] = src.ssavaluetypes[i]
                         end
-                        symtyps[lhsnode] = src.ssavaluetypes[i]
                     end
                     # Now process the RHS
                     stmt = stmt.args[2]
@@ -684,4 +673,15 @@ function getnextline(lt, j)
         j += 1
     end
     return typemax(Int)
+end
+
+function find_identifier_or_tuplechild(node, sym)
+    kind(node) == K"Identifier" && node.val == sym && return node, true
+    if kind(node) == K"tuple"   # tuple destructuring
+        for child in children(node)
+            tmpnode, found = find_identifier_or_tuplechild(child, sym)
+            found && return tmpnode, found
+        end
+    end
+    return node, false
 end

--- a/TypedSyntax/test/runtests.jl
+++ b/TypedSyntax/test/runtests.jl
@@ -164,6 +164,13 @@ include("test_module.jl")
     @test has_name_typ(child(t, 1), :x, Int)
     @test has_name_typ(child(t, 2), :y, Float64)
 
+    tsn = TypedSyntaxNode(TSN.bar381, (TSN.Foo381,))
+    sig, body = children(tsn)
+    lhs = child(body, 1, 1)
+    @test has_name_typ(child(lhs, 1), :a, Any)
+    @test has_name_typ(child(lhs, 2, 1), :b1, Any)
+    @test has_name_typ(child(lhs, 2, 2), :b2, Any)
+
     g = Base.Generator(identity, 1.0:4.0)
     tsn = TypedSyntaxNode(TSN.typeof_first_item, (typeof(g),))
     sig, body = children(tsn)
@@ -328,7 +335,6 @@ include("test_module.jl")
     @test Symbol("kwargs...") ∈ src.slotnames
     sig, body = children(tsn)
     @test child(body, 2, 1).typ <: Base.Pairs
-
 
     # Unused statements
     tsn = TypedSyntaxNode(TSN.mycheckbounds, (Vector{Int}, Int))

--- a/TypedSyntax/test/test_module.jl
+++ b/TypedSyntax/test/test_module.jl
@@ -117,4 +117,14 @@ fbroadcast_explicit(list) = sum(Base.materialize(Base.broadcasted(sin, list)))
 nospec(@nospecialize(x)) = 2x
 nospec2(@nospecialize(x::AbstractVecOrMat)) = first(x)
 
+# Nested tuple destructuring (issue #381)
+struct Foo381
+    a
+    b
+end
+function bar381(foo)
+    a, (b1, b2) = foo.a, foo.b
+    return b1
+end
+
 end

